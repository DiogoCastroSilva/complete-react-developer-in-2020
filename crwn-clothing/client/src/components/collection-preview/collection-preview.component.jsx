import React from 'react';
import { withRouter } from 'react-router-dom';

import CollectionItemContainer from '../../containers/collection-item.container';

import {
    CollectionPreviewContainer,
    TitleContainer,
    PreviewContainer
} from './collection-preview.styles';


export const CollectionPreview = ({
    title,
    items,
    history,
    match,
    routeName
}) => (
    <CollectionPreviewContainer>
        <TitleContainer
            onClick={() =>
                history.push(`${match.path}/${routeName}`)
            }
        >
            {title.toUpperCase()}
        </TitleContainer>
        <PreviewContainer>
            {items
                .filter((item, index) => index < 4)
                .map(item => (
                    <CollectionItemContainer
                        key={item.id}
                        item={item}
                    />
                ))
            }
        </PreviewContainer>
    </CollectionPreviewContainer>
);

export default withRouter(CollectionPreview);