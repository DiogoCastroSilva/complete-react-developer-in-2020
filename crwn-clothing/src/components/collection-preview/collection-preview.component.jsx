import React from 'react';

import './collection-preview.styles.scss';


const CollectionPreview = ({ title, items }) => (
    <div>
        <h1 className='collection-preview'>{title.toUpperCase()}</h1>
        <div className='preview'>
            {items
                .filter((_, index) => index < 4)
                .map(item => (
                    <div key={item.id}>{item.name}</div>
                ))
            }
        </div>
    </div>
);

export default CollectionPreview;