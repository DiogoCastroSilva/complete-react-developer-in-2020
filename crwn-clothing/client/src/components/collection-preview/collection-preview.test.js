import React from 'react';
import { shallow } from 'enzyme';

import { CollectionPreview } from './collection-preview.component';
import CollectionItem from '../collection-item/collection-item.component';
import { TitleContainer } from './collection-preview.styles';


describe('testing CollectionPreview component', () => {
    const mockRouteName = 'hats';
    let wrapper;
    let mockMatch;
    let mockHistory;

    beforeEach(() => {
        mockMatch = {
            path: '/shop'
        };

        mockHistory = {
            push: jest.fn()
        };

        const mockProps = {
            title: 'hats',
            items: [],
            history: mockHistory,
            match: mockMatch,
            routeName: mockRouteName
        };

        wrapper = shallow(<CollectionPreview {...mockProps} />);
    });

    it('should render a CollectionPreview component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should go back in history when clicked', () => {
        wrapper.find(TitleContainer).simulate('click');
        expect(mockHistory.push).toHaveBeenCalledWith(
            `${mockMatch.path}/${mockRouteName}`
        );
    });

    it('should render the right number of items', () => {
        const withItemsMockProps = {
            title: 'hats',
            items: [
                {
                    name: 'Brown Brim',
                    price: 25,
                    imageUrl: 'https://i.ibb.co/ZYW3VTp/brown-brim.png',
                    addItem: jest.fn()
                }
            ],
            history: mockHistory,
            match: mockMatch,
            routeName: mockRouteName
        };

        const newWrapper = shallow(<CollectionPreview {...withItemsMockProps} />);

        expect(newWrapper.find(CollectionItem).length).toEqual(withItemsMockProps.items.length);
    });
});