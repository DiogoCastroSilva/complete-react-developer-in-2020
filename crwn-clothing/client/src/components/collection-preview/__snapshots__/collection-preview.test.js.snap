// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CollectionPreview component should render a CollectionPreview component 1`] = `
<styled.div>
  <styled.h1
    onClick={[Function]}
  >
    HATS
  </styled.h1>
  <styled.div />
</styled.div>
`;
